pkg/scheduler/queue/tree_queueing_algorithms.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package queue
+
+// DequeueAlgorithm represents the set of operations specific to different approaches to dequeuing. It is applied
+// at the layer-level -- every Node at the same depth in a TreeQueue shares the same DequeueAlgorithm, including state
+// that may be stored in a struct that implements DequeueAlgorithm.
+type DequeueAlgorithm interface {
+	addChildNode(*Node, *Node)
+	deleteChildNode(*Node, *Node) bool
+	dequeueGetNode(*Node) (*Node, bool)
+	dequeueUpdateState(*Node, any, bool)
```

Comment:
can we name the args? Or otherwise select naming / write some docstrings that clarifies the purpose of the args?

I recognize that naming the args in the interface kind of restricts the implementations to no longer be able to choose the arg names, but I am mostly concerned about
`dequeueUpdateState(*Node, any, bool)` - if I want to implement this algorithm it's unclear to me what the point of each arg is.
As a consumer I can see one implementation calls the bool `nodeDeleted`, but another ignores it so it's not definitive.